test/test_is_freight_container.py

```diff
@@ -0,0 +1,18 @@
+import unittest
+from unittest import result
+
+from pyvalidator.is_freight_container import is_valid_container_number
+from . import print_test_ok
+
+class ContainerNumberValidationTest(unittest.TestCase):
+    def test_valid_container_number(self):
+        container_number = "ASDD123456"
```

Comment:
The validator is failing for this container ids, which are valid:

```
        'EMCU3811879',
        'OOLU8643084',
        'HJCU1922713',
```

Recheck the Regex.


I think the method needs a bit more changes, something like:

```
import re

isISO6346Str = re.compile(r'^[A-Z]{3}(U[0-9]{7})|([J,Z][0-9]{6,7})$')
isDigit = re.compile(r'^[0-9]$')

def isISO6346(str):
    assertString(str)
    str = str.upper()

    if not isISO6346Str.match(str):
        return False

    if len(str) == 11:
        sum = 0
        for i in range(len(str) - 1):
            if not isDigit.match(str[i]):
                convertedCode = None
                letterCode = ord(str[i]) - 55
                if letterCode < 11:
                    convertedCode = letterCode
                elif letterCode >= 11 and letterCode <= 20:
                    convertedCode = 12 + (letterCode % 11)
                elif letterCode >= 21 and letterCode <= 30:
                    convertedCode = 23 + (letterCode % 21)
                else:
                    convertedCode = 34 + (letterCode % 31)
                sum += convertedCode * (2 ** i)
            else:
                sum += int(str[i]) * (2 ** i)

        checkSumDigit = sum % 11
        return int(str[len(str) - 1]) == checkSumDigit

    return True
```